Use go to install latest obs-installer

vulkoingim opened the pull request:
## Description
<!-- Describe your changes in detail -->
https://github.com/gitpod-io/gitpod/pull/12989 fixed the issue with installing the obs-installer, so we no longer need to clone the repository anymore.

## Related Issue(s)
<!-- List the issue(s) this PR solves -->
Fixes #

## How to test
<!-- Provide steps to test this PR -->

## Release Notes
<!--
  Add entries for the CHANGELOG.md or "NONE" if there aren't any user facing changes.
  Each line becomes a separate entry.
  Format: [!<optional for breaking>] <description>
  Example: !basic auth is no longer supported
  See https://www.notion.so/gitpod/Release-Notes-513a74fdd23b4cb1b3b3aefb1d34a3e0
-->
```release-note
NONE
```

## Documentation
<!--
Does this PR require updates to the documentation at www.gitpod.io/docs?
* Yes
  * 1. Please create a docs issue: https://github.com/gitpod-io/website/issues/new?labels=documentation&template=DOCS-NEW-FEATURE.yml&title=%5BDocs+-+New+Feature%5D%3A+%3Cyour+feature+name+here%3E
  * 2. Paste the link to the docs issue below this comment
* No
  * Are you sure? If so, nothing to do here.
-->

## Werft options:
<!--
Optional annotations to add to the werft job.

* with-preview - whether to create a preview environment for this PR
-->
- [ ] /werft with-preview

mads-hartmann commented on .werft/observability/install-satellite.sh:
```diff
@@ -24,14 +24,11 @@ if ! command -v envsubst; then
-mkdir -p "${obsDir}"
-git clone https://roboquat:"$(cat /mnt/secrets/monitoring-satellite-preview-token/token)"@github.com/gitpod-io/observability.git "${obsDir}"
-cd "${obsDir}/installer"
+GOBIN=$(pwd) go install github.com/gitpod-io/observability/installer@main
```
would it be possible to have the binary be named "observability-installer" just to avoid confusing it with the Gitpod Installer.

Alternatively `rm ./installer` after line 31 would also work

mads-hartmann commented:
> started the job as [gitpod-build-aa-go-obs-installer.1](https://werft.gitpod-dev.com/job/gitpod-build-aa-go-obs-installer.1) because the annotations in the pull request description changed
(with .werft/ from main)

@vulkoingim To test this you'll need to run `werft job run github` to work around job protection

mads-hartmann reviewed (approved):
Changes look good to me. Added the hold label to give you time to test this. It also seems like the Werft build is broken for what looks like unrelated reasons

vulkoingim commented:
> Changes look good to me. Added the hold label to give you time to test this. It also seems like the Werft build is broken for what looks like unrelated reasons

I deleted the job, as I was running it manually.

This is a manual run: https://werft.gitpod-dev.com/job/gitpod-custom-aa-go-obs-installer.2

I'l retrigger so it allows the merge.

vulkoingim commented:
/werft run
